src/Random.mo

```diff
@@ -1,15 +1,87 @@
 /// Random number generation
 
 import Iter "Iter";
-import Prim "mo:⛔";
+import { todo; unreachable } "Debug";
 
 module {
 
   let rawRand = (actor "aaaaa-aa" : actor { raw_rand : () -> async Blob }).raw_rand;
 
   public let blob : shared () -> async Blob = rawRand;
 
-  // Remove `Finite` class?
+  /// Opinionated choice of PRNG from a given seed.
+  public func new(seed : Blob) : Random {
+    todo()
+  };
+
+  /// Uses entropy from the management canister with automatic resupply.
+  public func newAsync() : AsyncRandom {
+    var iter = Iter.empty<Nat8>();
+    AsyncRandom(
+      func() : async* Nat8 {
+        switch (iter.next()) {
+          case (?n) n;
+          case null {
+            iter := (await blob()).vals();
+            switch (iter.next()) {
+              case (?n) n;
+              case null unreachable()
+            }
+          }
+        }
+      }
+    )
+  };
+
+  public class Random(generator : () -> Nat8) {
+
+    /// Random choice between `true` and `false`.
+    public func bool() : Bool {
+      todo()
+    };
+
+    /// Random `Nat8` value in the range [0, 256).
+    public func byte() : Nat8 {
+      todo()
+    };
+
+    /// Random `Float` value in the range [0, 1).
```

Comment:
maybe mention that the resolution is maximal (all relevant mantissa bits are random)